File: app/src/main/java/com/shuoye/video/adapter/ViewHolder.java
package com.shuoye.video.adapter;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;

import org.jetbrains.annotations.NotNull;

import java.util.Map;

/**
 * @author shuoy
 */
public class ViewHolder<T extends ViewDataBinding> extends RecyclerView.ViewHolder {
    public T binding;
    public Map<String, Object> data;

    public ViewHolder(@NonNull @NotNull View itemView) {
        super(itemView);
    }
}

File: app/src/main/java/com/shuoye/video/adapters/ViewHolder.kt
package com.shuoye.video.adapters

import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView

/**
 * TODO
 * @program Video
 * @ClassName ViewHolder
 * @author shuoye
 * @create 2021-10-23 23:48
 **/


class ViewHolder<T : ViewDataBinding>(
    binding: T,
) : RecyclerView.ViewHolder(binding.root)